modules/videoio/src/cap_interface.hpp

```diff
@@ -221,6 +221,30 @@ class IVideoCapture
     virtual int getCaptureDomain() { return CAP_ANY; } // Return the type of the capture object: CAP_DSHOW, etc...
 };
 
+static void applyMetadataRotation(const IVideoCapture& cap, OutputArray mat)
```

Comment:
`rotation_auto` and `rotation_angle` are enough as parameters. I propose to split concerns.

Reply:
So, something like

```
 static void applyMetadataRotation(bool rotation_auto, int rotation_angle, OutputArray mat) {
  if (!rotation_auto) return;
  ...
}
```
and at the call site

```
applyMetadataRotation(
  0 != cap.getProperty(CAP_PROP_ORIENTATION_AUTO),
  static_cast<int>(cap.getProperty(CAP_PROP_ORIENTATION_META)), 
  image);
```
?

I don't what the call site looks like in this variation. Maybe split applyMetadataRotation in two:

```
static void applyMetadataRotation(const IVideoCapture& cap, OutputArray mat) {
  bool rotation_auto = 0 != cap.getProperty(CAP_PROP_ORIENTATION_AUTO);
  int rotation_angle = static_cast<int>(cap.getProperty(CAP_PROP_ORIENTATION_META));
  if (rotation_auto) {
    applyMetadataRotation(mat, rotation_angle);
  }
}

static void applyMetadataRotation(int rotation_angle, OutputArray mat) {
   ...
}
```

Reply:
Yes, I mean `static void applyMetadataRotation(int rotation_angle, bool autorotation, OutputArray mat)`. The current version works for me too.

Reply:
I'd prefer to keep it as-is, then. The next PR that adds support for orientation metadata to another backend can decide to refactor this function, I guess, if necessary.